src/valkey-cli.c

```diff
@@ -3469,12 +3469,27 @@ static int evalMode(int argc, char **argv) {
         }
         fclose(fp);
 
+        char *engine_name = NULL;
+        if (script[0] == '#' && script[1] == '!') {
+            const char *sp = strchr(script, '\n');
+            engine_name = strndup(script + 2, (sp - script) - 2);
```

Comment:
Engine name is only the first word in the shebang. There can be some parameters after the engine name, separated by a space.